diff --git a/drivers/modem/vendor_modem_cellular/cellular_sqn_gm02s.c b/drivers/modem/vendor_modem_cellular/cellular_sqn_gm02s.c
index 2768c6642a9..3d201a30c23 100644
--- a/drivers/modem/vendor_modem_cellular/cellular_sqn_gm02s.c
+++ b/drivers/modem/vendor_modem_cellular/cellular_sqn_gm02s.c
@@ -10,12 +10,17 @@
 
 MODEM_CELLULAR_COMMON_CHAT_MATCHES();
 
-MODEM_CHAT_MATCHES_DEFINE(sqn_gm02s_unsol, MODEM_CELLULAR_COMMON_UNSOL_MATCHES);
+MODEM_CHAT_MATCHES_DEFINE(sqn_gm02s_unsol,
+			  MODEM_CELLULAR_COMMON_UNSOL_MATCHES,
+			  MODEM_CHAT_MATCH("+SYSSTART", "", modem_cellular_chat_on_modem_ready));
 
 MODEM_CHAT_SCRIPT_CMDS_DEFINE(
-	sqn_gm02s_init_chat_script_cmds, MODEM_CHAT_SCRIPT_CMD_RESP("ATE0", ok_match),
+	sqn_gm02s_init_chat_script_cmds,
+	MODEM_CHAT_SCRIPT_CMD_RESP("ATE0", ok_match),
 	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CFUN=4", ok_match),
-	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CMEE=1", ok_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CPSMS=0", ok_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CEDRXS=0", ok_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CMEE=2", ok_match),
 	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CEREG=1", ok_match),
 	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CEREG?", ok_match),
 	MODEM_CHAT_SCRIPT_CMD_RESP("AT+CGSN", imei_match), MODEM_CHAT_SCRIPT_CMD_RESP("", ok_match),
@@ -54,8 +59,8 @@ static const struct modem_cellular_vendor_config sqn_gm02s_vendor = {
 	},
 	/* clang-format on */
 	.power_pulse_duration_ms = 1500,
-	.reset_pulse_duration_ms = 100,
-	.startup_time_ms = 2000,
+	.reset_pulse_duration_ms = 1000,
+	.startup_time_ms = 30000,
 	.shutdown_time_ms = 5000,
 	.force_autostart = true,
 };
--- a/drivers/modem/modem_cellular.c
+++ b/drivers/modem/modem_cellular.c
@@ -1472,9 +1472,12 @@
 	case MODEM_CELLULAR_EVENT_DEREGISTERED:
-		modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_AWAIT_PPP_DEAD);
+		if (!net_if_is_carrier_ok(modem_ppp_get_iface(data->ppp))) {
+			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_AWAIT_PPP_DEAD);
+		}
 		break;
 	case MODEM_CELLULAR_EVENT_PPP_DEAD:
-		if (net_if_is_admin_up(modem_ppp_get_iface(data->ppp))) {
+		if (net_if_is_admin_up(modem_ppp_get_iface(data->ppp)) &&
+		    !net_if_is_carrier_ok(modem_ppp_get_iface(data->ppp))) {
 			modem_cellular_enter_state(data, MODEM_CELLULAR_STATE_AWAIT_PPP_DEAD);
 			modem_cellular_start_timer(data, MODEM_CELLULAR_PERIODIC_SCRIPT_TIMEOUT);
 		}
 		break;
