diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75ece41..7cf75a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,16 @@ function(_rust_map_target)
     set(RUST_TARGET "x86_64-unknown-none" PARENT_SCOPE)
   elseif(CONFIG_ARCH_POSIX AND CONFIG_64BIT AND (${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "aarch64"))
     set(RUST_TARGET "aarch64-unknown-none" PARENT_SCOPE)
+  elseif(CONFIG_XTENSA)
+    if(CONFIG_SOC STREQUAL "esp32s3")
+      set(RUST_TARGET "xtensa-esp32s3-none-elf" PARENT_SCOPE)
+    elseif(CONFIG_SOC STREQUAL "esp32")
+      set(RUST_TARGET "xtensa-esp32-none-elf" PARENT_SCOPE)
+    elseif(CONFIG_SOC STREQUAL "esp32s2")
+      set(RUST_TARGET "xtensa-esp32s2-none-elf" PARENT_SCOPE)
+    else()
+      message(FATAL_ERROR "Rust: Unsupported Xtensa SOC")
+    endif()
   else()
     message(FATAL_ERROR "Rust: Add support for other target")
   endif()
@@ -68,6 +78,16 @@ function(get_include_dirs target dirs)
 endfunction()
 
 function(rust_cargo_application)
+  cmake_parse_arguments(RUST_APP "" "CRATE_NAME;FEATURES" "" ${ARGN})
+  if(NOT RUST_APP_CRATE_NAME)
+    set(RUST_APP_CRATE_NAME "rustapp")
+  endif()
+  if(RUST_APP_FEATURES)
+    set(rust_features_arg "--no-default-features" "--features" "${RUST_APP_FEATURES}")
+  else()
+    set(rust_features_arg "")
+  endif()
+
   # For now, hard-code the Zephyr crate directly here.  Once we have
   # more than one crate, these should be added by the modules
   # themselves.
@@ -94,7 +114,7 @@ function(rust_cargo_application)
   set(BUILD_LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/${RUST_TARGET}/${RUST_BUILD_TYPE}")
 
   set(CARGO_TARGET_DIR "${CMAKE_CURRENT_BINARY_DIR}/rust/target")
-  set(RUST_LIBRARY "${CARGO_TARGET_DIR}/${RUST_TARGET}/${RUST_BUILD_TYPE}/librustapp.a")
+  set(RUST_LIBRARY "${CARGO_TARGET_DIR}/${RUST_TARGET}/${RUST_BUILD_TYPE}/lib${RUST_APP_CRATE_NAME}.a")
   set(SAMPLE_CARGO_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/rust/sample-cargo-config.toml")
 
   # The generated C binding wrappers. These are bindgen-generated wrappers for the inline functions
@@ -173,14 +193,9 @@ ${config_paths}
       WRAPPER_FILE="${WRAPPER_FILE}"
       DT_AUGMENTS="${DT_AUGMENTS}"
       BINARY_DIR_INCLUDE_GENERATED="${BINARY_DIR_INCLUDE_GENERATED}"
-      cargo build
+      cargo $<$<BOOL:${CONFIG_XTENSA}>:+esp> build --lib
       ${rust_build_type_arg}
-
-      # Override the features according to the shield given. For a general case,
-      # this will need to come from a variable or argument.
-      # TODO: This needs to be passed in.
-      # --no-default-features
-      # --features ${SHIELD_FEATURE}
+      ${rust_features_arg}
 
       # Set a replacement so that packages can just use `zephyr-sys` as a package
       # name to find it.
@@ -192,7 +207,7 @@ ${config_paths}
   )
 
   # Be sure we don't try building this until all of the generated headers have been generated.
-  add_custom_target(librustapp ALL
+  add_custom_target(lib${RUST_APP_CRATE_NAME} ALL
     DEPENDS ${DUMMY_FILE}
         # The variables, defined at the top level, don't seem to be accessible here.
         offsets_h
@@ -253,7 +268,7 @@ ${config_paths}
   # Example of warning reported by ld when this fix is not in place:
   # <path>/ld.bfd: rust/target/riscv64imac-unknown-none-elf/debug/librustapp.a(45c91108d938afe8-clzdi2.o): unsupported relocation type 0x3d
   target_link_libraries(app PUBLIC $<TARGET_PROPERTY:linker,rt_library> -Wl,--allow-multiple-definition ${RUST_LIBRARY})
-  add_dependencies(app librustapp)
+  add_dependencies(app lib${RUST_APP_CRATE_NAME})
 
   # Presumably, Rust applications will have no C source files, but cmake will require them.
   # Add an empty file so that this will build.  The main will come from the rust library.
diff --git a/Kconfig b/Kconfig
index 0b2d057..6bebb7c 100644
--- a/Kconfig
+++ b/Kconfig
@@ -8,6 +8,7 @@ menu "Rust Language Support"
 config RUST_SUPPORTED
 	bool
 	default y if ((CPU_CORTEX_M || \
+				XTENSA || \
 				(RISCV && !RISCV_ISA_RV32E && !RISCV_ISA_RV128I) || \
 				(ARCH_POSIX && 64BIT)) && \
 				!TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
