diff --git a/subsys/modem/modem_cmux.c b/subsys/modem/modem_cmux.c
index 22d3b705cd4..8ee5df0b81d 100644
--- a/subsys/modem/modem_cmux.c
+++ b/subsys/modem/modem_cmux.c
@@ -140,7 +140,7 @@ static bool is_transitioning_to_powersave(struct modem_cmux *cmux)
 
 static bool is_connected(struct modem_cmux *cmux)
 {
-	bool connected;
+	bool connected = false;
 
 	K_SPINLOCK(&cmux->work_lock) {
 		if (!cmux->attached) {
@@ -156,7 +156,7 @@ static bool is_connected(struct modem_cmux *cmux)
 
 static bool is_connecting(struct modem_cmux *cmux)
 {
-	bool connecting;
+	bool connecting = false;
 
 	K_SPINLOCK(&cmux->work_lock) {
 		if (!cmux->attached) {
