diff --git a/zephyr/Cargo.toml b/zephyr/Cargo.toml
index 0cebba8..f995b81 100644
--- a/zephyr/Cargo.toml
+++ b/zephyr/Cargo.toml
@@ -56,15 +56,15 @@ version = "0.2"
 optional = true
 
 [dependencies.embassy-time-queue-utils]
-version = "0.1"
+version = "0.3"
 optional = true
 
 [dependencies.embassy-sync]
-version = "0.6.2"
+version = "0.7"
 optional = true
 
 [dependencies.embassy-executor]
-version = "0.7.0"
+version = "0.10"
 optional = true
 
 # These are needed at build time.
