diff --git a/CMakeLists.txt b/CMakeLists.txt
index be31379..d733314 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,16 @@ function(_rust_map_target)
     set(RUST_TARGET "x86_64-unknown-none" PARENT_SCOPE)
   elseif(CONFIG_ARCH_POSIX AND CONFIG_64BIT AND (${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "aarch64"))
     set(RUST_TARGET "aarch64-unknown-none" PARENT_SCOPE)
+  elseif(CONFIG_XTENSA)
+    if(CONFIG_SOC STREQUAL "esp32s3")
+      set(RUST_TARGET "xtensa-esp32s3-none-elf" PARENT_SCOPE)
+    elseif(CONFIG_SOC STREQUAL "esp32")
+      set(RUST_TARGET "xtensa-esp32-none-elf" PARENT_SCOPE)
+    elseif(CONFIG_SOC STREQUAL "esp32s2")
+      set(RUST_TARGET "xtensa-esp32s2-none-elf" PARENT_SCOPE)
+    else()
+      message(FATAL_ERROR "Rust: Unsupported Xtensa SOC")
+    endif()
   else()
     message(FATAL_ERROR "Rust: Add support for other target")
   endif()
@@ -104,7 +114,7 @@ function(add_cargo_target_with_zephyr_env target)
       WRAPPER_FILE="${WRAPPER_FILE}"
       DT_AUGMENTS="${DT_AUGMENTS}"
       BINARY_DIR_INCLUDE_GENERATED="${BINARY_DIR_INCLUDE_GENERATED}"
-      cargo ${cargo_command}
+      cargo $<$<BOOL:${CONFIG_XTENSA}>:+esp> ${cargo_command}
       ${rust_build_type_arg}
 
       # Set a replacement so that packages can just use `zephyr-sys` as a package
@@ -128,6 +138,16 @@ function(add_cargo_target_with_zephyr_env target)
 endfunction()
 
 function(rust_cargo_application)
+  cmake_parse_arguments(RUST_APP "" "CRATE_NAME;FEATURES" "" ${ARGN})
+  if(NOT RUST_APP_CRATE_NAME)
+    set(RUST_APP_CRATE_NAME "rustapp")
+  endif()
+  if(RUST_APP_FEATURES)
+    set(rust_features_arg "--no-default-features" "--features" "${RUST_APP_FEATURES}")
+  else()
+    set(rust_features_arg "")
+  endif()
+
   # For now, hard-code the Zephyr crate directly here.  Once we have
   # more than one crate, these should be added by the modules
   # themselves.
@@ -154,7 +174,7 @@ function(rust_cargo_application)
   set(BUILD_LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/${RUST_TARGET}/${RUST_BUILD_TYPE}")
 
   set(CARGO_TARGET_DIR "${CMAKE_CURRENT_BINARY_DIR}/rust/target")
-  set(RUST_LIBRARY "${CARGO_TARGET_DIR}/${RUST_TARGET}/${RUST_BUILD_TYPE}/librustapp.a")
+  set(RUST_LIBRARY "${CARGO_TARGET_DIR}/${RUST_TARGET}/${RUST_BUILD_TYPE}/lib${RUST_APP_CRATE_NAME}.a")
   set(SAMPLE_CARGO_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/rust/sample-cargo-config.toml")
 
   # The generated C binding wrappers. These are bindgen-generated wrappers for the inline functions
@@ -213,26 +233,14 @@ ${config_paths}
 ")
 
   # The library is built by invoking Cargo.
-  add_cargo_target_with_zephyr_env(librustapp
+  add_cargo_target_with_zephyr_env(lib${RUST_APP_CRATE_NAME}
     ALL
     OUTPUT ${DUMMY_FILE}
     BYPRODUCTS ${RUST_LIBRARY} ${WRAPPER_FILE}
-    CARGO_ARGS build
+    CARGO_ARGS build --lib ${rust_features_arg}
     COMMENT "Building Rust application"
   )
 
-  # Command to lint the Rust application.
-  add_cargo_target_with_zephyr_env(clippy
-    ALL
-    OUTPUT run_rust_clippy
-    CARGO_ARGS
-      clippy
-      --
-      -D warnings
-      -D clippy::undocumented_unsafe_blocks
-    COMMENT "Linting Rust application"
-  )
-
   # Command to generate the rust docs.
   add_cargo_target_with_zephyr_env(rustdoc
     OUTPUT generate_rust_docs
@@ -249,7 +257,7 @@ ${config_paths}
   # Example of warning reported by ld when this fix is not in place:
   # <path>/ld.bfd: rust/target/riscv64imac-unknown-none-elf/debug/librustapp.a(45c91108d938afe8-clzdi2.o): unsupported relocation type 0x3d
   target_link_libraries(app PUBLIC $<TARGET_PROPERTY:linker,rt_library> -Wl,--allow-multiple-definition ${RUST_LIBRARY})
-  add_dependencies(app librustapp)
+  add_dependencies(app lib${RUST_APP_CRATE_NAME})
 
   # Presumably, Rust applications will have no C source files, but cmake will require them.
   # Add an empty file so that this will build.  The main will come from the rust library.
diff --git a/Kconfig b/Kconfig
index 0b2d057..6bebb7c 100644
--- a/Kconfig
+++ b/Kconfig
@@ -8,6 +8,7 @@ menu "Rust Language Support"
 config RUST_SUPPORTED
 	bool
 	default y if ((CPU_CORTEX_M || \
+				XTENSA || \
 				(RISCV && !RISCV_ISA_RV32E && !RISCV_ISA_RV128I) || \
 				(ARCH_POSIX && 64BIT)) && \
 				!TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
diff --git a/zephyr-build/src/lib.rs b/zephyr-build/src/lib.rs
index 6fd4f77..a3fa2b3 100644
--- a/zephyr-build/src/lib.rs
+++ b/zephyr-build/src/lib.rs
@@ -77,7 +77,7 @@ pub fn build_kconfig_mod() {
     for line in BufReader::new(file).lines() {
         let line = line.expect("reading line from dotconfig");
         if let Some(caps) = config_hex.captures(&line) {
-            writeln!(&mut f, "#[allow(dead_code)]").unwrap();
+            writeln!(&mut f, "#[allow(dead_code, overflowing_literals)]").unwrap();
             writeln!(&mut f, "pub const {}: usize = {};", &caps[1], &caps[2]).unwrap();
         } else if let Some(caps) = config_int.captures(&line) {
             writeln!(&mut f, "#[allow(dead_code)]").unwrap();
